docs/nv2/README.md

```diff
@@ -0,0 +1,376 @@
+  -o hello-world.signature.config.json \
+  file:hello-world_v1-manifest.json
+```
+
+The formatted x509, without the `x5c` chain signature: `hello-world.signature.config.json` is:
+
+
+```json
+{
+    "signed": {
+        "iat": 1595456231,
+        "manifests": [
+            {
+                "digest": "sha256:407a722870b09ef1c037b3bd9d1e6fa828a1c64964ba8c292a8ebe4dcf3bde56",
+                "size": 3056,
+                "references": [
+                    "acme-rockets.io/hello-world:v1"
+                ]
+            }
+        ]
+
+    },
+    "signatures": [
+        {
+            "typ": "x509",
+            "sig": "FyRRcAGi1qd0IHT8XoRh0vlGSkE4rjYpJzYEjodRQ2aUO0O/bBIzjV86UxqLnb1Y/GMU817YXeqHqDlySLWYoGKg+/aJGJDqbQpWIxUr6hhjGaxBYDZTt2ayzMAu5X/GNCx0vRLKl5dOOsgTO53QbuKEf4F3xxvQJv3rXHnObJbPSPCavxzs5TNRLepYEZzW1Mp5nkZT4l32/7QLnwwzTsJYGOMTmhGZ7O5LB/eeViKmwBJHXpNzd4rytFXccKlPuyUakSKgsPdTjEvY5UbFpH568wG21HXDQivz6qdST9eSVob2yUx7WV7z+2S2GfmiMZ30BMtKs4Jx1uPOY3Hk8g==",
+            "alg": "RS256",
+            "kid": "2MKO:CS4G:GP3F:HELH:TUI2:5YSX:NJNU:3O2N:LYM4:FBHC:T7NN:OM5A"
+
+        }
+    ]
+}
+```
+
+Within the signature, the claims  `alg`, `x5c`, `kid` are specified by [RFC 7515](https://tools.ietf.org/html/rfc7515)
+
+### Signing using GnuPG
+
+To sign the manifest `example.json` using the GnuPG key identified by the identity name `Demo User`, run
+
+``` shell
+nv2 sign -m gpg \
+  -i "Demo User" \
+  -r registry.acme-rockets.io/hello-world:v1 \
+  -e 8760h \
+  -o hello-world.signature.config.json \
+  file:hello-world_v1-manifest.json
+```
+
+- `-r` declares the original registry reference - which must match the key.
+- `-e` specifies the optional expiry time (`8760h = 365 days`)
+
+On successful signing, `nv2` prints out the `sha256` digest of the manifest, and writes the `nv2` signature JSON file `<digest>.nv2` to the working directory. If the file name is not desired, option `-o` can be specified for the alternative file name.
+
+The formatted signature file is:
+
+```json
+{
+  "signed": {
+    "exp": 1626792407,
+    "nbf": 1595256407,
+    "iat": 1595256407,
+    "manifests": [
+      {
+        "digest": "sha256:3351c53952446db17d21b86cfe5829ae70f823aff5d410fbf09dff820a39ab55",
+        "size": 528,
+        "references": [
+          "example.registry.io/example:latest"
+        ]
+      }
+    ]
+  },
+  "signatures": [
+    {
+      "typ": "gpg",
+      "sig": "wsDcBAABCAAQBQJfFa5XCRBGnsnNNqPeHgAAlM0MAHFXZxyCgsxiGVat8YCRIhR7IoQe2scswGyvGGinYBy88EpKGFEAGO+Kt1frTQNW9kLYWmTw4EFctgMw+XxeDD/CI2rsMSluRh9h2t8xBsO9Ux+7eJoxSEsfU8Jc/YZWpGs/kJOGQ3ERjvPt+SCG0Y8tuNtjnzpV4Gz+8fLSlNZ7b3f+rd7nvvJuB8iWr+yojsCeWh/VGuibyqAXPKVxSrKgkmziyYK3O/0D3KhgyR+CMtjTXL5hP314Gpc415YyN82LC3L44okimN/+X3avX0vQkthiyVw+R+Vgmpa1qk1P/ySrs81yQgFBPBC7+m4n54TqsW46X/UlkQdfP/x5Jg3jUURKgQb0wSLvzbr7Jk1RiThlwjcLhM0VgRIUwbqcqjg/5UNvMRehD44PxQXRz5feZjER2awMyKqRZnImpm8Ub+hAjhqtLGYT34oU2lwctoObV4f4BzffY9kQ0x37PQ3V8aj8k6YFQZbB4vLgwtZdA2c1froVHyuRBUwLzSBevg==",
```

Comment:
for reference, here are the subpackets *inside* this signature:

```
# off=0 ctb=c2 tag=2 hlen=3 plen=412 new-ctb
:signature packet: algo 1, keyid 469EC9CD36A3DE1E
	version 4, created 1595256407, md5len 0, sigclass 0x00
	digest algo 8, begin of digest 94 cd
	hashed subpkt 2 len 4 (sig created 2020-07-20)
	hashed subpkt 16 len 8 (issuer key ID 469EC9CD36A3DE1E)
	data: [3071 bits]
```

I'm curious, has there been any exploration as to what:
- PGP signature classes will be supported?
- What digest algorithms will be supported?
- If a hashed subpacket of type 2 will be compared against either `exp` or  `iat` fields?
- If a fingerprint field will be recommended/checked? long keyids have been shown to be collidable with consumer-grade computing power

Reply:
https://evil32.com/

Reply:
We've been debating the value of gpg. The initial PR had gpg support to demonstrate multiple signing methods. We pulled it out as it seemed to add some confusion that it could bypass the x509 cert. 
What do folks think about gpg support?

Reply:
My understanding is that the current PR is more about showing the extensibility to allow crypto providers to plug in the right places. If other providers can't be plugged in then it would good to actually identify how the interface needs to expand to enable other algorithms and  signature formats to be supported.